src/aspire/numeric/pyfftw_fft.py

```diff
@@ -129,9 +131,15 @@ def fftn(self, a, axes=None, workers=-1):
         return b
 
     def ifftn(self, a, axes=None, workers=-1):
-        mutex.acquire()
 
-        comp_type = a.dtype
+        axes = axes or tuple(range(a.ndim))
+
+        # FFTW_BACKWARD requires complex input array, cast as needed.
+        # See https://pyfftw.readthedocs.io/en/latest/source/pyfftw/pyfftw.html#scheme-table
+        comp_type = complex_type(a.dtype)
+        a = a.astype(comp_type, copy=False)
```

Comment:
Do we want to modify the input array?

Reply:
In the existing code, we can, but you are totally right, that is dangerous. I'll patch this. Thanks!

Reply:
Wait, nm, I don't think this case occurs.  Are you sure?

Reply:
You mean we always input a complex type array? It's not the worst thing if the array I put in ends up complex but with the same data. Was just a thought

Reply:
No, that isn't what I mean.

What is being modified?

Reply:
if `a.dtype` is a real dtype

```
        comp_type = complex_type(a.dtype)
        a = a.astype(comp_type, copy=False)
```

Reply:
is `a` not having its dtype changed?

Reply:
which a? the one on the right or left of the equals?

Reply:
Oh I see, you're right. Just a relabeling. Nvm